### DISCUSSION TITLE
Add Thingspeak support

### DISCUSSION THREAD

**Discussion by flhoest** [jomjol/AI-on-the-edge-device#2321] [Ideas]
Hi There,

I just discovered this amazing project, I'm busy building it, it would be awesome to have the ability to connect natively the data to a ThingSpeak channel.

Thanks !

Fred.

**Comment by flhoest** [comment]
and of course send the gaps and not the counter value. ;)

**Comment by RobbesU** [comment]
Hi @flhoest, did you manage to connect AI-on-the-Edge to Thingspeak?  I have an already existing project loading to TS, and have just now added an AIothED. It's working and connecting to TS but cannot send the correct payload (i.e. fieldX= [main/value])

**Reply by WWillyWacker** [reply]
Hi RobbesU, I am currently facing the same challenge of mapping the values read by AI-on-the-edge to the configured channel name and a field in the channel. I have already created an MQTT device called ‘watermeter’ in Thingspeak, but I am not sure how to map it to a field in the authorised channel. Have you found anything helpful on this?

**Comment by caco3** [comment]
I don't like to use a company driven cloud to store/publish/process my data.
Why not using your own Home Automation or write to your local InfluxDB?

Looking at https://blogs.mathworks.com/iot/2017/01/20/use-mqtt-to-send-iot-data-to-thingspeak/ it should be possible to use MQTT to send the data to Thinkspeak

**Comment by RobbesU** [comment]
Very roundabout: Because I didn't have time to mess with my ThingSpeak, I ended up feeding the MQTT from the camera device to a local HomeAssistant (which is baked into the device) and then used that to push it to my ThingSpeak from there... in the end too much hassle so past tense. 

Or you could dig into the firmware code and add the ability yourself :-)
https://www.mathworks.com/help/thingspeak/use-arduino-client-to-publish-to-a-channel.html
